runtime/tr.source/trj9/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -9778,154 +9778,127 @@ static TR::Register * inlineSinglePrecisionSQRT(TR::Node *node, TR::CodeGenerato
-
-      cg->stopUsingRegister(tmpReg);
       }
    else
       {
       targetRegister = cg->allocateConsecutiveRegisterPair();
-      TR::Register * targetLowRegister  = targetRegister->getLowOrder();
-      TR::Register * targetHighRegister = targetRegister->getHighOrder();
 
-      TR::Register * tmpReg  = cg->allocateRegister();;
-      TR::Register * tmpReg2 = cg->allocateRegister();;
-
-      int32_t offsets[3];
+      TR::Register* tempRegister1 = cg->allocateRegister();
+      TR::Register* tempRegister2 = cg->allocateRegister();
 
 #if defined(TR_HOST_S390) && defined(J9ZOS390)
-      _getSTCKLSOOffset((int32_t*)&offsets);
+      int32_t offsets[3];
+      _getSTCKLSOOffset(offsets);
 
-      // zOS requires some time correction to account for Leap Seconds.
-      // The number of leap seconds is stored in the LSO field of the
-      // MVS data area.
+      // z/OS requires time correction to account for leap seconds. The number of leap seconds is stored in the LSO
+      // field of the MVS data area.
       if (TR::Compiler->target.isZOS())
          {
          // Load FFCVT(r0)
-         TR::MemoryReference *tmpMemRef = generateS390MemoryReference(tmpReg, offsets[0],cg);
-         tmpMemRef->setBaseRegister(NULL, cg);
-         generateRXYInstruction(cg, TR::InstOpCode::L, node, tmpReg, tmpMemRef);
+         generateRXYInstruction(cg, TR::InstOpCode::L, node, tempRegister1, generateS390MemoryReference(offsets[0], cg));
+
          // Load CVTEXT2 - CVT
-         generateRXYInstruction(cg, TR::InstOpCode::L, node, tmpReg, generateS390MemoryReference(tmpReg, offsets[1],cg));
-         // Subtract the LSO Offset.
+         generateRXYInstruction(cg, TR::InstOpCode::L, node, tempRegister1, generateS390MemoryReference(tempRegister1, offsets[1],cg));
          }
 #endif
 
-      TR::MemoryReference * lhref = generateS390MemoryReference(node, symRef, cg);
-      generateRSInstruction(cg, TR::InstOpCode::LM, node, targetRegister, lhref);
+      generateRSInstruction(cg, TR::InstOpCode::LM, node, targetRegister, generateS390MemoryReference(node, reusableTempSlot, cg));
 
-      int32_t todJan1970H = 0x7D91048B;
-      int32_t todJan1970L = 0xCA000000;
+      int32_t todJanuary1970High = 0x7D91048B;
+      int32_t todJanuary1970Low = 0xCA000000;
 
-      // tmpReg2 is not actually used
-      //
-      generateS390ImmOp(cg, TR::InstOpCode::SL, node, tmpReg2, targetLowRegister, todJan1970L, NULL, litBase);
-      generateS390ImmOp(cg, TR::InstOpCode::SLB, node, tmpReg2, targetHighRegister, todJan1970H, NULL, litBase);
+      // tempRegister2 is not actually used
+      generateS390ImmOp(cg, TR::InstOpCode::SL, node, tempRegister2, targetRegister->getLowOrder(), todJanuary1970Low, NULL, literalBaseRegister);
+      generateS390ImmOp(cg, TR::InstOpCode::SLB, node, tempRegister2, targetRegister->getHighOrder(), todJanuary1970High, NULL, literalBaseRegister);
```

Comment:
I had concerns that on 31-bit we were not actually subtracting 1970 but this is not the case. The `generateS390ImmOp` API aliases the source with the target register for `SL` and `SLB`. Everything should work as expected.